In [1]:
import heapq
from collections import Counter

In [2]:
#FILE = "example.txt"
FILE = "input.txt"

In [3]:
inputs = [
    list(line.split()) for line in open(FILE)
]

In [4]:
inputs

[['463765957799961-468948402182307'],
 ['514282982719515-514535485734829'],
 ['484765627298551-486228972785924'],
 ['515168098763186-515339396365033'],
 ['306348303280454-309851686438206'],
 ['150870242160173-151671240285641'],
 ['446558215591126-450159213845213'],
 ['172002071120812-179067001590563'],
 ['71750732200219-73189398343452'],
 ['254841833346550-260031247506485'],
 ['94327557076326-94792660654059'],
 ['536594929104644-540864094433780'],
 ['302297185747946-306348303280453'],
 ['25620701070989-27712728684461'],
 ['489997739127463-491439182040439'],
 ['151357843823034-151671240285641'],
 ['3478808518345-7804660665544'],
 ['404924036056066-408157696638263'],
 ['490657168920149-492266952469168'],
 ['93278418174390-94055622593837'],
 ['319029810026759-321560548683434'],
 ['181610633794455-186973892121936'],
 ['262275597324019-266842953366361'],
 ['374683527225097-375251279181585'],
 ['95634958254905-96297636385592'],
 ['473269468318174-482156876004959'],
 ['517915404520645-5185023

In [5]:
ranges = []
while inputs:
    line = inputs.pop(0)
    if not line:
        break
    ranges.extend(line)
ranges = [(int(i.split('-')[0]),int(i.split('-')[1])) for i in ranges]

In [6]:
ranges = sorted(ranges, key=lambda x: x[0])
#ranges

In [7]:
len(ranges)

185

In [8]:
numbers = [int(i[0]) for i in inputs]

In [9]:
numbers.sort()

In [15]:
fresh_count = 0
freshes = []
# for number in numbers:
#     for start,end in ranges:
#         if start <= number <= end:
#             break
#     else:
#         spoiled_count += 1
for number in numbers:
    spoiled = True
    for start,end in ranges:
        if start <= number <= end:
            #print(f"{number} is fresh at range {start}-{end}")
            freshes.append(number)
            spoiled = False
            break
    if not spoiled:
        fresh_count += 1

In [16]:
fresh_count

520

480 was too low

520 was right, I was counting the spoiled items